test/e2e/framework.go

```diff
@@ -970,6 +970,17 @@ func (data *TestData) createNginxPodOnNode(name string, nodeName string) error {
 	}, false, nil)
 }
 
+// createNginxPodWithPort creates an nginx Pod in the test namespace with configurable container port.
+func (data *TestData) createNginxPodWithPort(name, nodeName string, port int32) error {
+	return data.createPodOnNode(name, nodeName, nginxImage, []string{}, nil, nil, []corev1.ContainerPort{
+		{
+			Name:          "http",
+			ContainerPort: port,
+			Protocol:      corev1.ProtocolTCP,
+		},
+	}, false, nil)
+}
```

Comment:
this function should be moved to `nodeportlocal_test.go` IMO. But I am not sure why we even need it. You always use the same port number (90) and I see no reference to the port name ("http").